imports/wasi_snapshot_preview1/fs_test.go

```diff
@@ -1104,10 +1105,27 @@ func Test_fdReaddir(t *testing.T) {
 
 				return &internalsys.FileEntry{File: dir}
 			},
-			buf: 0, bufLen: 24, // length is too short
+			buf: 0, bufLen: 23, // length is too short for header
 			cookie:          0,
-			expectedBufused: 24, // == bufLen which is the size of the dirent
+			expectedBufused: 23, // == bufLen which is the size of the dirent
 			expectedMem:     nil,
+			expectedReadDir: &internalsys.ReadDir{
+				CountRead: 2,
+				Entries:   testDirEntries[:2],
+			},
+		},
+		{
+			name: "can't read name",
```

Comment:
I have added this test, because this is probably actually what the test above wanted to test? This tests if we can write away the header when we have exactly 24 bytes in the buffer.